File: 04_Managing_Indexes/04_QueryLSI.cs
client = new AmazonDynamoDBClient();

QueryRequest queryRequest = new QueryRequest
{
    TableName = "product",
    IndexName = "IdManufacturerIndex",
    Select = "ALL_ATTRIBUTES",
    ScanIndexForward = true,
    KeyConditionExpression = "id = :v_id AND mnfr = :v_mnfr",
    ExpressionAttributeValues = new Dictionary<string,
    AttributeValue>()
   {
        {":v_id",new AttributeValue {N = "1"}},
        {":v_mnfr",new AttributeValue {S = "samsung"}}
    },
};

var result = client.Query(queryRequest);

File: 04_Managing_Indexes/04_queryLSI.js
var params = {
    TableName: 'productTableLSI',
    IndexName: "IdManufacturerIndex",
    Select: 'ALL_ATTRIBUTES',
    ScanIndexForward : true,
    KeyConditionExpression: 'id = :v_id AND mnfr = :v_mnfr', // a string representing a constraint on the attribute
    ExpressionAttributeValues: { // a map of substitutions for all attribute values
      ':v_id': 1,
      ':v_mnfr': 'samsung'
    }
};
docClient.query(params, function(err, data) {
    if (err) ppJson(err); // an error occurred
    else ppJson(data); // successful response

});